File: python/leetcode/rotate-image.py
class Solution:
    def rotate(self, matrix):
        """
        :type matrix: List[List[int]]
        :rtype: void Do not return anything, modify matrix in-place instead.
        """
        temp = 0
        for c in range(len(matrix)):
            for r in range(len(matrix[c])):
                if c < r:
                    # can short hand this with x, y = y, x assignment
                    # Just being more explicit and language agnostic
                    temp = matrix[c][r]
                    matrix[c][r] = matrix[r][c]
                    matrix[r][c] = temp
        for c in range(len(matrix)):
            for r in range(len(matrix[c]) // 2):
                temp = matrix[c][r]
                matrix[c][r] = matrix[c][len(matrix[c]) - 1- r]
                matrix[c][len(matrix[c]) - 1- r] = temp

File: java/leetcode/rotate-image.java
class Solution {
    public void rotate(int[][] matrix) {
        int temp = 0;
        for (int c = 0; c < matrix.length; c++) {
            for (int r = 0; r < matrix[c].length; r++) {
                if (c < r) {
                    temp = matrix[c][r];
                    matrix[c][r] = matrix[r][c];
                    matrix[r][c] = temp;
                }
            }
        }
        for (int c = 0; c < matrix.length; c++) {
            for (int r = 0; r < Math.round(matrix[c].length /2); r++) {
                temp = matrix[c][r];
                matrix[c][r] = matrix[c][matrix[c].length - 1- r];
                matrix[c][matrix[c].length - 1 - r] = temp;              
            }
        }
    }
}
